apps/03TicTacToe.js

```diff
@@ -24,35 +43,105 @@ function printBoard() {
 }
 
 function horizontalWin() {
-  // Your code here
+   //check for horizontal win
+  if ((board [0][0] === playerTurn) && (board [0][1] === playerTurn) && (board [0][2] === playerTurn)) {
+    return true;
+  }
+  if ((board [1][0] === playerTurn) && (board [1][1] === playerTurn) && (board [1][2] === playerTurn)) {
+    return true;
+  }
+  if ((board [2][0] === playerTurn) && (board [2][1] === playerTurn) && (board [2][2] === playerTurn)) {
+    return true;
+  }
 }
 
 function verticalWin() {
-  // Your code here
+  //check for vertical win
+  if ((board [0][0] === playerTurn) && (board [1][0] === playerTurn) && (board [2][0] === playerTurn)) {
+    return true;
+  }
+  if ((board [0][1] === playerTurn) && (board [1][1] === playerTurn) && (board [2][1] === playerTurn)) {
+    return true;
+  }
+  if ((board [0][2] === playerTurn) && (board [1][2] === playerTurn) && (board [2][2] === playerTurn)) {
+    return true;
+  }
 }
 
 function diagonalWin() {
-  // Your code here
+  //check for diagonal win
+  if ((board [0][0] === playerTurn) && (board [1][1] === playerTurn) && (board [2][2] === playerTurn)) {
+    return true;
+  }
+  if ((board [0][2] === playerTurn) && (board [1][1] === playerTurn) && (board [2][0] === playerTurn)) {
+    return true;
+  }
 }
 
 function checkForWin() {
-  // Your code here
+  if (horizontalWin() || verticalWin() || diagonalWin()) {
+    console.log('\nPlayer ' + playerTurn + ' Won!');
+    detectWin = true;
+    return true;
+  }
 }
 
+function checkForTie() {
+  var count = 0;
+  for (var i=0;i<3;i++) {
+    for (var j=0;j<3;j++) {
+      if (board[i][j] !== ' ') {
+        count += 1;
```

Comment:
If you increment your `counter` variable every time a valid move is made, you can shorten your checkForTie() function by running only lines 98-100.